modules/lfs/http_client.go

```diff
@@ -17,19 +18,22 @@ import (
 	"code.gitea.io/gitea/modules/proxy"
 )
 
-const batchSize = 20
+const httpBatchSize = 20
+
+var ErrUnexpectedEOF = errors.New("unexpected EOF in response")
 
 // HTTPClient is used to communicate with the LFS server
 // https://github.com/git-lfs/git-lfs/blob/main/docs/api/batch.md
 type HTTPClient struct {
-	client    *http.Client
-	endpoint  string
-	transfers map[string]TransferAdapter
+	client        *http.Client
+	endpoint      string
+	transfers     map[string]TransferAdapter
+	transferNames []string
```

Comment:
Why it needs this separate `transferNames` field? If I understand correctly, the `transferNames` are the keys of the `transfers` map?